File: 평강/app/project/src/main/java/com/eventpurchase/project/domain/product/controller/ProductController.java
package com.eventpurchase.project.domain.product.controller;

import com.eventpurchase.project.domain.product.dto.response.ProductResponse;
import com.eventpurchase.project.domain.product.service.ProductService;
import com.eventpurchase.project.global.common.response.CursorResponse;
import com.eventpurchase.project.global.common.response.SuccessResponse;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Slf4j
@RestController
@RequestMapping("/api/product")
@RequiredArgsConstructor
public class ProductController {

    private final ProductService productService;


    @GetMapping
    public ResponseEntity<SuccessResponse<CursorResponse<ProductResponse>>> getProducts(
            @RequestParam(required = false) Long lastId,
            @RequestParam(defaultValue = "20") int size)
    {
        CursorResponse<ProductResponse> products = productService.getProducts(lastId, size);
        return ResponseEntity.ok(SuccessResponse.of(HttpStatus.OK, "상품 목록 조회 성공", products));
    }

    @GetMapping("/{id}")
    public ResponseEntity<SuccessResponse<ProductResponse>> getProduct(@PathVariable Long id) {
        ProductResponse productResponse = productService.getProductById(id);
        return ResponseEntity.ok(SuccessResponse.of(HttpStatus.OK, "상품 상세 조회 성공", productResponse));
    }

}

File: 평강/app/project/product/api/src/main/kotlin/com/eventpurchase/project/product/ProductController.kt
package com.eventpurchase.project.product

import com.eventpurchase.project.shared.response.CursorResponse
import com.eventpurchase.project.shared.response.SuccessResponse
import org.springframework.web.bind.annotation.*
import java.math.BigDecimal

@RestController
@RequestMapping("/api/products")
class ProductController(private val productService: ProductService) {

    @GetMapping
    fun getProducts(
        @RequestParam(required = false) lastId: Long?,
        @RequestParam(defaultValue = "20") size: Int
    ): SuccessResponse<CursorResponse<ProductResponse>> {
        val result = productService.getProducts(lastId, size)
        return SuccessResponse.of(200, "상품 목록 조회 성공",
            CursorResponse(result.content.map { it.toResponse() }, result.nextLastId, result.hasNext))
    }

    @GetMapping("/{id}")
    fun getProduct(@PathVariable id: Long): SuccessResponse<ProductResponse> =
        SuccessResponse.of(200, "상품 상세 조회 성공", productService.getProductById(id).toResponse())
}

data class ProductResponse(
    val id: Long?, val productName: String, val brand: String, val category: String,
    val subCategory: String?, val price: BigDecimal, val discountRate: BigDecimal?,
    val rating: BigDecimal?, val reviewCount: Int?, val imageUrl: String?, val source: String?
)

private fun Product.toResponse() = ProductResponse(id, productName, brand, category, subCategory, price, discountRate, rating, reviewCount, imageUrl, source)
